src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneStorageTable/CloneStorageTable.test.tsx

```diff
@@ -8,60 +6,62 @@ import {
   nodeFilesystem as fsFactory,
   nodePartition as partitionFactory,
 } from "testing/factories";
+import { render, screen, within } from "testing/utils";
 
 describe("CloneStorageTable", () => {
   it("renders empty table if neither loading details nor machine provided", () => {
-    const wrapper = mount(
-      <CloneStorageTable machine={null} selected={false} />
-    );
-    expect(wrapper.find("MainTable").prop("rows")).toStrictEqual([]);
-    expect(wrapper.find("Placeholder").exists()).toBe(false);
+    render(<CloneStorageTable machine={null} selected={false} />);
+    expect(screen.getAllByRole("row")).toHaveLength(1);
+    expect(screen.queryByTestId("placeholder")).not.toBeInTheDocument();
   });
 
   it("renders placeholder content while machine details are loading", () => {
-    const wrapper = mount(
+    render(
       <CloneStorageTable
         loadingMachineDetails
         machine={null}
         selected={false}
       />
     );
-    expect(wrapper.find("Placeholder").exists()).toBe(true);
+    const rows = screen.getAllByRole("row");
+    rows.shift();
```

Comment:
`shift` mutates the original array, it's best to stick with immutable methods that don't cause unexpected side effects by changing the original array.

It seems like the intention here is to target rows within the table body and skip the table header. You can also do that by targeting the second row group of a table, e.g.

```tsx
const tableBody = screen.getAllByRole("rowgroup")[1];
within(tableBody) [...]
```